go-controller/pkg/ovn/loadbalancer/loadbalancer.go

```diff
@@ -1,332 +1,300 @@
 package loadbalancer
 
 import (
-	"encoding/json"
 	"fmt"
+	"sort"
 	"strings"
 
-	"github.com/pkg/errors"
-	utilnet "k8s.io/utils/net"
+	"k8s.io/apimachinery/pkg/util/sets"
 
-	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/types"
 	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
 
-	kapi "k8s.io/api/core/v1"
 	"k8s.io/klog/v2"
 )
 
-const (
-	// used to indicate if the service is running on node load balancers
-	NodeLoadBalancer = "NodeLoadBalancer"
-	// used to indicate if the service is running on idling load balancers
-	IdlingLoadBalancer = "IdlingLoadBalancer"
-)
+// EnsureLBs provides a generic load-balancer reconciliation engine.
+//
+// It assures that, for a given set of ExternalIDs, only the configured
+// list of load balancers exist. Existing load-balancers will be updated,
+// new ones will be reated as needed, and stale ones will be deleted.
+//
+// For example, you might want to ensure that service ns/foo has the
+// correct set of load balancers. You would call it with something like
+//
+//     EnsureLBs( { kind: Service, owner: ns/foo}, { {Name: Service_ns/foo_cluster_tcp, ...}})
+//
+// This will ensure that, for this given Service, only that one load balancer exists and
```

Comment:
nit: "that only the correct loadbalancer/s exist/s"

Reply:
this applies to the above example, clarified.